python_modules/libraries/dagster-shared/dagster_shared/record/__init__.py

```diff
@@ -618,17 +621,3 @@ def _pydantic_core_schema(cls, source: Any, handler):
     from pydantic_core import core_schema
 
     return core_schema.is_instance_schema(cls)
-
-
-TRecord = TypeVar("TRecord")
-
-
-class NamedTupleAdapter(Generic[TRecord]):
```

Comment:
im fine using this name if we prefer it

Reply:
I'm used to the mixin at this point, and it makes enough sense to me. Also like that it says "legacy" in the name to discourage new usage